server/src/main/java/org/elasticsearch/reservedstate/service/FileSettingsService.java

```diff
@@ -146,9 +172,23 @@ private void processFileChanges(ReservedStateVersionCheck versionCheck) throws I
         completion.get();
     }
 
+    private void completeProcessing(Exception e, PlainActionFuture<Void> completion) {
+        if (e != null) {
+            completion.onFailure(e);
```

Comment:
Oh now I remember why. The failure incrementing happens in [onProcessFileChangesException](https://github.com/elastic/elasticsearch/pull/117081/files#diff-5a2c0417a5b4bdbcc6873fc853db7a8c531b47536bd41b84e20bb6c7b21ad6d9R185) because then we have some flexibility to handle it in different ways based on the nature of the exception.

I need to ponder this a bit. At the very least, this deserves an explanatory comment; but probably I should rearrange it so it makes more sense.

Reply:
I think the consensus in our discussion was to report these as `YELLOW` unconditionally, and use the `Details` to differentiate how we alert on them.